Viết chương trình Python sử dụng một trong 2 loại vòng lặp for hoặc while để
tính tống các số nguyên từ 1 đến n. (n <= 1000)

In [3]:
n=int(input())
print(n*(n+1)//2)

28


In [4]:
n=int(input())
s=0
for i in range(1,n+1):
    s+=i
print(s)

55


In [6]:
n=int(input())
s=0
while n>0:
    s+=n
    n-=1
print(s)

28


VIết chương trình nhập vào một số tới khi được một số nguyên dương thì thôi.
Kiểm tra số này có phải là số nguyên tố hay không? 


In [23]:
def nhap():
    n=int(input())
    if n>0:return n
    else:
        return nhap()
def nto(n):
    if n<2:return False
    if n in [2,3,5,7]: return True
    if (n%2==0) or (n%3==0) or (n%5==0) or (n%7==0):return False
    i=5
    while i*i<=n:
        i+=6
        if (n%i==0) or (n%(i+2)==0):return False
    return True
k=nhap()
if (nto(k)):print(f"Day la so nguyen to")
else :print("Day khong la so nguyen to")

Day khong la so nguyen to


In [22]:
def nhap():
    n=int(input())
    if n>0:return n
    else:
        return nhap()
def nto(n):
    if n<2:return False
    for i in range(2,int(n**0.5)+1):
        if (n%i==0):return False
    return True
k=nhap()
if (nto(k)):print(f"Day la so nguyen to")
else :print("Day khong la so nguyen to")

Day la so nguyen to


In [ ]:
def nhap():
    n=int(input())
    if n>0:return n
    else:
        return nhap()
def sang():
    
    a[0]=0
    a[1]=0
    for i in range(2,int(1000000**0.5)+1):
        if a[i]==1:
            for j in range(i**2,1000000,i):
                a[j]=0
k=nhap()
a=[1]*(1000000)
sang()
if a[k]==1:print(f"Day la so nguyen to")
else :print("Day khong la so nguyen to")

Day la so nguyen to


Viết chương trình tính giai thừa của một số nguyên n cho trước (n!=1×2×…×n)
với 0 < n < 100. 


In [33]:
n=int(input())
t=1
for i in range(1,n+1):
    t=t*i
print(t)

120


In [36]:
n=int(input())
t=1
k=1
while k<=n:
    t=t*k
    k+=1
print(t)

120


Viết chương trình nhập vào một số nguyên n và in ra màn hình số chữ số (trừ dấu)
của số đó. Không dùng kiểu dữ liệu string.


In [40]:
n=int(input())
if n<0:n=n*-1
d=1
while n>=10:
    n=n//10
    d+=1
print(d)

7


Khi Douglas Adam viết cuốn sách, The Hitchhiker’s Guide to the Galaxy, ông ta đã
thêm vào đó một vài lời nói đùa, và theo năm tháng nó đã trở nên nổi tiếng hơn cả
bản thân cuốn tiểu thuyết: “Câu trả lời cho câu hỏi lớn nhất về sự sống, về vũ trụ và
tất cả mọi thứ là 42.” Một vài người cuồng tín đã tốn hàng năm trời, với mọi cố
gắng nỗ lực để gán ghép ý nghĩa, biểu trưng cho con số 42 và các “sự cố” liên quan
đến nó.
Với mong muốn hái ra tiền từ sư “ám ảnh” điên rồ này, một cuốn sách mới vừa
được xuất bản, 42: Douglas Adams’ Amazingly Accurate Answer to Life, the
Universe and Everything. Cuốn sách được phát hành trùng với kỉ niệm 10 năm ngày
mất của Adams.
“Câu trả lời rất đơn giản”, như Adams đã nói, “Chỉ là một trò đùa. Tôi cần một con
số, một con số be bé, một con số tầm thường, và tôi đã chọn nó. Hai chữ số, bội của
13, một con số hoàn toàn vô nghĩa đối với mấy tay thầy chùa. Tôi ngồi trên bàn làm
việc, ngắm khu vườn và nghĩ rằng số 42 sẽ được việc lắm đây. Tôi gõ nó ra. Và
hết.”
Xuyên suốt lịch sử, các con số được gán cho những ý nghĩa đặc biệt nào đó. Plato
gọi việc nghiên cứu các biểu tượng số học” là level cao nhất của tri thức. Trong khi
Pythagoras tin rằng các con số đều có linh hồn cũng như sức mạnh kì diệu.
Trong khi đó, hàng triệu người hâm hộ Hitchhiker cho đến ngày nay vẫn kiên trì
giải mã những thứ họ cho rằng đó là những động cơ bí mật của Adams.
Hãy viết một chương trình nhận vào một số nguyên dương n và một dãy gồm n số
nguyên. Nếu n số nguyên đó có số 42 thì in ra màn hình dòng chữ "I′ve found the
meaning of life!", ngược lại in ra dòng chữ "It′s a joke!".


In [42]:
n=int(input())
a=list(map(int,input().split()))
for i in a:
    if i==42:
        k="I′ve found the meaning of life!"
        break
    else:
        k="It′s a joke!"
print(k)

I′ve found the meaning of life!


Nhập hai số nguyên dương a, b (a ≤ b). Hãy tính tổng các số nguyên tố trong đoạn
[a, b]. Mỗi dòng là một bộ test (a b)."


In [49]:
n,m=map(int,input().split())
a=[1]*(m+2)
a[0]=0
a[1]=0
s=0
for i in range(2,int(m**0.5)+1):
    if a[i]==1:
        for j in range(i*i,m+1,i):
            a[j]=0
for i in range(n,m+1):
    if a[i]==1:s=s+i
print(s)


98


Nhập vào số nguyên dương n (n ≥ 2). Hãy in ra ước số nguyên tố lớn nhất của n.

In [51]:
n=int(input())
k=2
for i in range(2,n+1):
    while n%i==0:
        n=n//i
        k=max(k,i)
print(k)

7


Nhập số nguyên dương n. Lặp lại phép toán n = n + reverse(n) cho đến khi n là số
đối xứng (palindrome). In ra số bước và giá trị palindrome thu được

In [ ]:
n=int(input())
def sodao(n):
    cs=0
    while n>0:
        cs=cs*10+n%10
        n=n//10
    return cs
def dx(n):
    if n==(sodao(n)):return True
    return False
d=0
while not dx(n):
    n=n+sodao(n)
    d+=1
print(d,n)

786
1473
5214
9339
4 9339


Nhập vào số nguyên dương n. In ra tất cả các số chính phương ≤ n có các chữ số
đôi một khác nhau (in theo thứ tự tăng dần, cùng hàng, cách nhau bởi một dấu
cách). Nếu không có, in ra (no number).


In [119]:
n=int(input())
def cp(n):
    if int(n**0.5)*int(n**0.5)==n:return True
    return False
def dx(n):
    i=0
    a=[]
    while n>0:
        a.append(n%10)
        n=n//10
    for i in range (0,len(a)):
        for j in range(0,len(a)-i-1):
            if a[j]<a[j+1]:
                a[j],a[j+1]=a[j+1],a[j]
    for i in range(0,len(a)-1):
        if a[i]==a[i+1]:
            i=1
            break
    if i!=1:    return True
    else:return False
d=0
for i in range(1,n+1):
    if cp(i) and dx(i):
        d+=1
        print(i)
if d==0:print("no number")

1
4
9
16
25
36
49
64
81
1024
1089
1296
1369
1764
1849
1936
2304
2401
2601
2704
2809
2916
3025
3249
3481
3721
4096
4356
4761
5041
5184
5329
5476
6084
6241
6724
7056
7396
7569
7921
8649
9025
9216
9604
9801
12769
13456
13689
13924
15376
15876
16384
17689
17956
18496
18769
20164
20736
21609
21904
23104
23409
23716
28561
29584
30276
30625
30976
31684
32041
32761
34596
35721
36481
37249
38025
38416
39204
39601
41209
43681
45369
45796
47089
47961
50176
51076
51984
53824
54289
61504
63504
67081
71289
71824
72361
73984
74529
78961
79524
81796
82369
83521
85264
87025
89401
91204
92416
93025
95481
96721
103684
104329
104976
107584
123904
124609
132496
134689
139876
140625
157609
162409
164025
170569
173056
180625
195364
198025
207936
209764
214369
237169
254016
257049
258064
259081
279841
293764
310249
318096
321489
326041
327184
329476
341056
346921
349281
351649
356409
357604
379456
381924
385641
390625
391876
395641
401956
405769
408321
416025
418609
423801
436921
458329
470596
483025
497025
5

Nhập vào số nguyên dương n. Tìm số x trong [1, n] có độ dài dãy Collatz lớn nhất
và in ra x cùng độ dài dãy tương ứng (x L). Nếu có nhiều số cùng độ dài, chọn số
nhỏ nhất.

In [12]:
n=int(input())
def kt(n):
    d=0
    while n!=1:
        if n%2==0:n=n//2
        else: n=3*n+1
        d+=1
    return d
def pr(n):
    d=0
    while n!=1:
        if n%2==0:n=n//2
        else: n=3*n+1
        d+=1
        print(n,end=" ")
    return d
m=1
j=1
for i in range(2,n+1):
    if kt(i)>m:
        m=kt(i)
        j=i
print(j)
pr(j)

97
292 146 73 220 110 55 166 83 250 125 376 188 94 47 142 71 214 107 322 161 484 242 121 364 182 91 274 137 412 206 103 310 155 466 233 700 350 175 526 263 790 395 1186 593 1780 890 445 1336 668 334 167 502 251 754 377 1132 566 283 850 425 1276 638 319 958 479 1438 719 2158 1079 3238 1619 4858 2429 7288 3644 1822 911 2734 1367 4102 2051 6154 3077 9232 4616 2308 1154 577 1732 866 433 1300 650 325 976 488 244 122 61 184 92 46 23 70 35 106 53 160 80 40 20 10 5 16 8 4 2 1 

118

Viết chương trình đếm số lượng ước số chẵn của số nguyên dương n với n < 10^6.

In [31]:
n=int(input())
a={}
for i in range(2,n):
    while n%i==0:
        a[i]=a.get(i,0)+1
        n=n//i
t=a[2]
for i in a:
    if i!=2:
        t=t*(a[i]+1)
print(t)

3


Một người có tài khoản tiết kiệm ở ngân hàng và gửi vào X đồng với lãi suất là
0.7% mỗi tháng. Viết chương trình tính số tiền sau N tháng người ấy rút được (cả
gốc và lãi, bỏ qua phần lẻ thập phân).

In [ ]:
x,n=map(float,input().split())
print(f"{x*(1+0.7*0.01)**n:.0f}")

2
2.466167337481387


Hai số nguyên dương tạo thành một cặp số thân thiết khi chúng tuân theo quy luật:
Số này bằng tổng tất cả các ước của số kia (trừ chính nó) và ngược lại.
Ví dụ 220 và 284 là một cặp số thân thiết
220 ngoài chính nó ra, nó còn có 11 ước số khác là: 1, 2, 4, 5, 10, 11, 20, 44, 55,
110. Tổng của 11 ước này cho chúng ta 284
284 ngoài bản thân nó, còn có 5 ước số khác là: 1, 2, 4, 71, 142. Tổng của chúng
vừa đúng bằng 220
Nhập hai số từ bàn phím và kiểm tra xem chúng có phải là cặp số thân thiết không.
Nếu có, in ra màn hình "true", ngược lại, in ra "false" 


In [39]:
n,m=map(int,input().split())
def kt(n):
    s=0
    for i in range (1,n):
        if n%i==0:
            s=s+i
    return s
if kt(n)==m and kt(m)==n:print("True")
else:print("False")

True


Cho hai số m và n là hai số nguyên dương nhập từ bàn phím. Viết chương trình
tìm ước chung lớn nhất của hai số m và n. 


In [42]:
n,m=map(int,input().split())
def ucln(n,m):
    while m!=0:
        n,m=m,n%m
    return abs(n)
print(ucln(n,m))

12


[ChickenAndDog]
"Vừa gà vừa chó,
Bó lại cho tròn,
Ba mươi sáu con,
Một trăm chân chẵn"
Từ bài toán dân gian trên, mở rộng thành chương trình nhận đầu vào là tổng số
con và tổng số chân của gà và chó. Nếu tìm được số phù hợp, in ra số lượng gà và
số lượng chó. Ngược lại, in ra "invalid"

In [ ]:
n,m=map(int,input().split())
y=-1
if (m-2*n)%2==0:
    y=(m-2*n)//2
if y>0:
    x=n-y
    print(x,y )
else:print("invalid") 

22 14
